src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1512,6 +1512,7 @@ void GenerateResetMessage()
         internal void Reset()
         {
             Debug.Assert(State == ConnectorState.Ready);
+            ReadBuffer.ResetStatistics();
```

Comment:
Can this be placed behind a flag for `EnableStatistics`? What does this cost if `EnableStatistics` is `false`?